SPECS/reaper/reaper_build_caches.sh

```diff
@@ -0,0 +1,142 @@
+#! /bin/bash
+
+set -eux pipefail
+
+# NOTE: EXECUTING with id=0 (root) CAUSES REAPER BUILD TO FAIL.
+# Cassandra reaper version for which to generate build caches.
+# Set this to the required version.
+VERSION="3.1.1"
+# URL to download the sources to build
+SOURCE_URL="https://github.com/thelastpickle/cassandra-reaper/archive/refs/tags/${VERSION}.tar.gz"
+
+# Build cache names
+BOWER_COMPONENTS="reaper-bower-components-${VERSION}.tar.gz"
+SRC_UI_NODE_MODULES="reaper-srcui-node-modules-${VERSION}.tar.gz"
+BOWER_CACHE="reaper-bower-cache-${VERSION}.tar.gz"
+MAVEN_CACHE="reaper-m2-cache-${VERSION}.tar.gz"
+NPM_CACHE="reaper-npm-cache-${VERSION}.tar.gz"
+LOCAL_LIB_NODE_MODULES="reaper-local-lib-node-modules-${VERSION}.tar.gz"
+LOCAL_N="reaper-local-n-${VERSION}.tar.gz"
+
+tempDir=$(mktemp -d)
+
+mkdir -p ${tempDir}/mariner_caches
+
+marinerCacheDir=${tempDir}/mariner_caches
+
+function checkInternet {
+	sudo tdnf install -y nc
+	nc -z 8.8.8.8 443
+	if [ $? -eq 0 ]; then
+		echo "Online."
+	else
+		echo "Offline, reaper needs internet connectivity to download and build cache."
+		exit 1
+	fi
+}
+
+function dieIfError {
+	if [ $? -ne 0 ];then
+		echo "Error $?, exiting build."
+		exit $?
+	fi
+}
+
+function installNodeModules {
+	echo "Installing node modules."
+	sudo tdnf install -y nodejs | dieIfError
+	# Default node/npm versions in Mariner fails to build dependency node module versions due to known
+	# incompatibilities.
+	# Backward compatible with node@v14.18.0
+	# When installing modules via npm to default prefix='/usr/local' in mariner, the permissions for 'others'
+	# is incoorectly set that causes 'which' to still point to older path, as access/newfstatat fail with -ENOPERM
+	# Setting a new global npm folder for fixing permission issues.
+	# (works well with id=0, but reaper build will fail.)
+	mkdir --mode 0777 $HOME/.npm-global
+	echo $HOME
+	npm config set prefix "$HOME/.npm-global"
+	echo $HOME
```

Comment:
Debugging leftover? What are these `echo $HOME` for?

Reply:
Removed.